# HG changeset patch
# User J. Ryan Stinnett <jryans@gmail.com>
# Date 1441085078 18000
#      Tue Sep 01 00:24:38 2015 -0500
# Node ID 8985273b4676f06d2cdd3fb7049ab43d5d112f57
# Parent  4525f2b8d431f6ae0159c6861978b6e4f328a976
Bug 912121 - Create shims for popular DevTools themes in add-ons. r=bgrins

For popular themes used by many DevTools add-ons, add shim files which wrap the
themes and make them available at their previous location.

diff --git a/devtools/client/moz.build b/devtools/client/moz.build
--- a/devtools/client/moz.build
+++ b/devtools/client/moz.build
@@ -32,16 +32,20 @@ DIRS += [
     'styleeditor',
     'styleinspector',
     'tilt',
     'webaudioeditor',
     'webconsole',
     'webide',
 ]
 
+# Shim old theme paths used by DevTools add-ons
+if CONFIG['MOZ_BUILD_APP'] == 'browser':
+    DIRS += ['themes/shims']
+
 EXTRA_COMPONENTS += [
     'devtools-clhandler.js',
     'devtools-clhandler.manifest',
 ]
 
 JAR_MANIFESTS += ['jar.mn']
 
 DevToolsModules(
diff --git a/devtools/client/themes/shims/common.css b/devtools/client/themes/shims/common.css
new file mode 100644
--- /dev/null
+++ b/devtools/client/themes/shims/common.css
@@ -0,0 +1,10 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+ /**
+  * This file only exists to support add-ons which import this style sheet at a
+  * specific path.
+  */
+
+@import url("chrome://devtools/skin/themes/common.css");
diff --git a/devtools/client/themes/shims/jar.mn b/devtools/client/themes/shims/jar.mn
new file mode 100644
--- /dev/null
+++ b/devtools/client/themes/shims/jar.mn
@@ -0,0 +1,6 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+browser.jar:
+    skin/classic/browser/devtools/common.css (common.css)
diff --git a/devtools/client/themes/shims/moz.build b/devtools/client/themes/shims/moz.build
new file mode 100644
--- /dev/null
+++ b/devtools/client/themes/shims/moz.build
@@ -0,0 +1,9 @@
+# -*- Mode: python; c-basic-offset: 4; indent-tabs-mode: nil; tab-width: 40 -*-
+# vim: set filetype=python:
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+# Shim old theme paths used by DevTools add-ons
+if CONFIG['MOZ_BUILD_APP'] == 'browser':
+    JAR_MANIFESTS += ['jar.mn']
